adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenRequestTest.java

```diff
@@ -558,6 +599,433 @@ public void testVerifyBrokerRedirectUriInvalidSignature() throws PackageManager.
     }
 
 
+    /**
+     * Positive Test for Resiliency Feature
+     *
+     * If
+     * 1. outage mode is on, which means ExtendedLifetimeEnabled is set to be true in AuthenticationContext
+     * 2. the token cache item is expired but not expired in the extended expire time
+     * 3. the server is down where we get the 500,503,504 error after the retry
+     * Then
+     * The authentication result with stale access token should be returned.
+     */
+    @SmallTest
+    public void testResiliencyTokenReturn_outageModeOn_positive() throws PackageManager.NameNotFoundException,
+            NoSuchAlgorithmException, OperationCanceledException, IOException, AuthenticatorException,
+            InterruptedException {
+        // make sure AT's expires_in is expired and ext_expires_in is not expired
+        final Calendar expiredTime = new GregorianCalendar();
+        expiredTime.add(Calendar.MINUTE, -MINUS_MINUITE);
+        final Calendar extendedExpiresTime = new GregorianCalendar();
+        extendedExpiresTime.add(Calendar.MINUTE, EXTEND_MINUS_MINUTE);
+        final ITokenCacheStore cacheStore = getExtendedLifetimeTokenCache(expiredTime.getTime(), false, false, extendedExpiresTime.getTime());
+
+        final AccountManager mockedAccountManager = getMockedAccountManager();
+        mockAccountManagerGetAccountBehavior(mockedAccountManager);
+        final FileMockContext mockContext = createMockContext();
+        mockContext.setMockedAccountManager(mockedAccountManager);
+        //set useBroker into false
+        AuthenticationSettings.INSTANCE.setUseBroker(false);
```

Comment:
Do you really need to mock the account manager? It's not the purpose of the test, I don't think there is any need to do it.